### ISSUE TITLE
[BUG] `component.licenses` not valid against schema in version 1.0

### ISSUE THREAD

**Issue by madpah** [CycloneDX/cyclonedx-python-lib#151]
Example code to produce BOM:
```python
bom = Bom(
        components=[Component(
            name='setuptools', version='50.3.2', bom_ref='pkg:pypi/setuptools@50.3.2?extension=tar.gz',
            purl=PackageURL(
                type='pypi', name='setuptools', version='50.3.2', qualifiers='extension=tar.gz'
            ), license_str='MIT License', author='Test Author'
        )]
    )
```

This attempts to produce XML for the Component as follows:
```xml
<component type="library" bom-ref="pkg:pypi/setuptools@50.3.2?extension=tar.gz">
            <name>setuptools</name>
            <version>50.3.2</version>
            <licenses>
                <expression>MIT License</expression>
            </licenses>
            <cpe>cpe:2.3:a:python:setuptools:50.3.2:*:*:*:*:*:*:*</cpe>
            <purl>pkg:pypi/setuptools@50.3.2?extension=tar.gz</purl>
        </component>
```

`.licenses.expression` is not permissible in schema version 1.0.

**Comment by jkowalleck** [comment]
@madpah  the fix u would suggest: drop support for spec 1.0 :-) 
spec 1.0 has a lot of missing elements, some assumptions that did not meat the real world ....

**Comment by madpah** [comment]
Agree that at some point @jkowalleck we need to drop support for `1.0` and possibly other old schema versions.

Will raise a ticket to track this.
